Update repo and script READMEs (#235)
AlexsLemonade/scpca-paper-figures: README.md

### Instructons for Data Lab members

To prepare data for figure and table generation, you will need to run the figure setup scripts:

```sh
Rscript scripts/figure_setup/sync-metadata.R
Rscript scripts/figure_setup/sync-data-files.R
Rscript scripts/figure_setup/sync-reference-files.R
Rscript scripts/figure_setup/sync-consensus-celltype-results.R
```

If you have setup `1Password` to handle your AWS credentials, you will need to prefix scripts beginning with `sync-` with `op run --`, specifically:

```sh
op run -- Rscript scripts/figure_setup/sync-metadata.R
op run -- Rscript scripts/figure_setup/sync-data-files.R
op run -- Rscript scripts/figure_setup/sync-reference-files.R
```

## Additional repository contents

### Sample info

The `sample-info/` folder contains metadata files used to create figures and tables.

* `brain-classifications-no-multiplexed.tsv`: This tsv file classifies brain-related diagnoses in the ScPCA Portal into "High-grade glioma" and "Low-grade glioma" for plotting.

> * `celltype-reference-metadata.tsv`: This tsv file contains information about references used for CellAssign cell type annotation on the ScPCA Portal.

Reviewer: ```suggestion
* `celltype-reference-metadata.tsv`: This tsv file contains information about references used for CellAssign and SingleR cell type annotation on the ScPCA Portal.
```
Okay, let's keep both files but specify that this one also includes SingleR/Celldex information to distinguish it from the file below.

## Generating figures and tables

The `figures/` and `tables/` folders contain the most up-to-date version of each of the figures and tables, respectively.
The `scripts/` folder contains all scripts used to create the figures and tables.
See the [`README` for the `scripts` folder](./scripts/README.md) for more information on figure and table scripts.

The [`generate-figures-tables.sh`](generate-figures-tables.sh) script can be used to prepare all figures and tables.
The script assumes that a folder called `s3_files` has been populated with relevant data files.
Therefore, **before running this script** you will first need to prepare these input data files, as described in the sections below.

Then, run the figure generation script as:

```sh
bash generate-figures-tables.sh
```

### Instructions to prepare data

> If you are not a member of the Data Lab, please follow the instructions provided in `reproduce-figures-analysis/obtain-prepare-data.md` to obtain and prepare additional data needed to regenerate figures and tables.

Reviewer: Just noting that we need to update this line with whatever folder name we decided on.

### Instructons for Data Lab members

> To prepare data for figure and table generation, will need to run the figure setup scripts:

Reviewer: ```suggestion
To prepare data for figure and table generation, you will need to run the figure setup scripts:
```

```sh
Rscript scripts/figure_setup/sync-metadata.R
Rscript scripts/figure_setup/sync-data-files.R
Rscript scripts/figure_setup/sync-reference-files.R

> Rscript scripts/figure_setup/assign-brain-classifications.R
> Rscript scripts/figure_setup/prepare-sample-whitelist.R

Reviewer: ```suggestion
Rscript scripts/figure_setup/sync-consensus-celltype-results.R
```
The output from these are stored in the repo so they doesn't need to be re-run. 
But we do need to sync the consensus results, so that should be added.

```

If you have setup `1Password` to handle your AWS credentials, you will need to prefix scripts beginning with `sync-` with `op run --`, specifically:

```sh
op run -- Rscript scripts/figure_setup/sync-metadata.R
op run -- Rscript scripts/figure_setup/sync-data-files.R
op run -- Rscript scripts/figure_setup/sync-reference-files.R
```

## Additional repository contents

### Sample info

The `sample-info/` folder contains metadata files used to create figures and tables.

* `brain-classifications-no-multiplexed.tsv`: This tsv file classifies brain-related diagnoses in the ScPCA Portal into "High-grade glioma" and "Low-grade glioma" for plotting.

> * `celltype-reference-metadata.tsv`: This tsv file contains information about references used for CellAssign cell type annotation on the ScPCA Portal.

Reviewer: I feel like with the addition of `scpca-project-celltype-metadata.tsv` we don't use this file anymore so we should get rid of it?

Author: It's still used here: https://github.com/AlexsLemonade/scpca-paper-figures/blob/ebb7c3ddcbdbb61a94ca965b3e544bcb61880adc/scripts/TableS2_cellassign-ref-summary.R#L13-L15

This file has overall descriptions of the references, but the other one is only names & project. We could combine the files into one probably, but I don't think that's necessary. Code works, good enough!

For each `submitted_disease_timing`, a `standardized_disease_timing` is assigned.
* `project-whitelist.txt`: This file contains a list of all projects that are currently active on the ScPCA Portal.
* `sample-whitelist.txt`: This file contains a list of all samples that are currently active on the ScPCA Portal.
* `scpca-project-celltype-metadata.tsv`: This tsv file provides the specific CellAssign cell typing reference used for each ScPCA project.

### Color palettes

The `palettes/` folder contains any palettes used in generating the figures.

* `diagnosis-group-palette.tsv`: This is the palette used to color the `diagnosis_group` for each sample.
* `disease-timing-palette.tsv`: This is the palette used to color the `disease_timing` for each sample.
* `immune-palette.tsv`: This is the palette used to color certain immune cell types from the overall consensus cell types.
* `suspension-palette.tsv`: This is the palette used to color libraries by `Single-cell` or `Single-nuclei`.
* `method-palette.tsv`: This is the palette used to color by quantification method used, either `Alevin-fry` or `Cell Ranger`.

> * `validation-group-palette.tsv`: This is the palette used to color validation groups used to assess consensus cell types.

Reviewer: ```suggestion
* `validation-group-palette.tsv`: This is the palette used to color broad cell type annotations used to assess consensus cell types.
```
Just because we don't really output anything that says validation group, we should be clear what colors in the figures this matches up to.
